server/src/domain/library/library.service.ts

```diff
@@ -188,25 +189,37 @@ export class LibraryService extends EventEmitter {
   }
 
   async getStatistics(auth: AuthDto, id: string): Promise<LibraryStatsResponseDto> {
-    await this.access.requirePermission(auth, Permission.LIBRARY_READ, id);
+    if (!auth.user.isAdmin) {
+      await this.access.requirePermission(auth, Permission.LIBRARY_READ, id);
+    }
```

Comment:
You should move this logic into the `requirePermission` implementation. It should return true if the user is an admin or execute the check otherwise. This is access control logic though, and that should exist inside of the access.core.ts file.

Reply:
Done